### PR TITLE
Bugfix/spi workaround calling sdk wrapper mock archive issue

### PR DESCRIPTION
**Pull request by kevinyulu**
## Purpose

* The recent introduction of using custom callingSDKWrapper in UITest caused issue in archiving the demo app. This PR includes a workaround for resolving the issue in archiving.

Issue:
Screen Shot 2022-11-23 at 2 17 39 PM

Cause: 
Swift Package Manager currently does not support the use of @testable import in archive, resulting in compiler not able to find reference of the module. The workaround is to expose the custom init function from CallComposite and its underling CallingSDKWrapperProtocol types by using @_spi attribute so that they can be imported by using SPI import. 

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull request checklist

This PR has considered:
- [ ] Color Theming
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] View Data Injection
- [ ] Demo App UI/UX update

## How to Test

* Open...
* Click on...

## What to Check

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Utilities/CallCompositeInternalError.swift [hunk 1]
```diff
@@ -59,8 +59,9 @@ enum CallCompositeInternalError: Error, Equatable {
     }
 }
 
-extension CallCompositeInternalError {
-    static func == (lhs: CallCompositeInternalError, rhs: CallCompositeInternalError) -> Bool {
+@_spi(CallCompositeUITest) public  extension CallCompositeInternalError {
+    @_spi(CallCompositeUITest) static func == (lhs: CallCompositeInternalError,
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Utilities/CallCompositeInternalError.swift:63]: I'm wondering if `public` is missed here

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/DI/DependencyContainer.swift [hunk 2]
```diff
@@ -32,11 +32,13 @@ final class DependencyContainer {
 
     func registerDependencies(_ callConfiguration: CallConfiguration,
                               localOptions: LocalOptions?,
-                              callCompositeEventsHandler: CallComposite.Events) {
+                              callCompositeEventsHandler: CallComposite.Events,
+                              withCallingSDKWrapper wrapper: CallingSDKWrapperProtocol? = nil) {
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/DI/DependencyContainer.swift:36]: `with` can be removed as other parameters doesn't include it

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Service/Calling/AzureCommunicationCalling/CallingSDKEventsHandler.swift [hunk 3]
```diff
@@ -147,14 +147,14 @@ class CallingSDKEventsHandler: NSObject, CallingSDKEventsHandling {
     }
 }
 
-extension CallingSDKEventsHandler: CallDelegate,
+@_spi(CallCompositeUITest) extension CallingSDKEventsHandler: CallDelegate,
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Service/Calling/AzureCommunicationCalling/CallingSDKEventsHandler.swift:150]: is`public` missed here?

Comment (kevinyulu) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Service/Calling/AzureCommunicationCalling/CallingSDKEventsHandler.swift:150]: this is by swift I think. swift does not allow us to specify access level on extension that declare protocol conformances. 
`'public' modifier cannot be used with extensions that declare protocol conformances`

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Utilities/CallCompositeInternalError.swift (hunk 1 above)
Comment (kevinyulu) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Utilities/CallCompositeInternalError.swift:63]: yeah good call. Actually swift does it for us in this case. adding `public` would have this warning so I removed the public 
`'public' modifier is redundant for operator function declared in a public extension`

**Comment by emlynmac** [comment]
I think given how much needs to be made public, we should probably just ensure that the changes for test can only be used with a debug build.
This would mean wrapping imports with #if DEBUG and corresponding functionality for the tests.

**Pull request closed without merging**